ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+# Orleans and Akka Actors: A Comparison
+
+## Step-By-Step Review
+
+This overview walks along the Technical Report [MSR-TR-2014-41](http://research.microsoft.com/apps/pubs/default.aspx?id=210931) by *Philip A. Bernstein, Sergey Bykov, Allan Geller, Cabriel Kliot, Jorgen Thelin (Microsoft Research)*, for another comparison focusing on Erlang and Riak see [Christopher Meiklejohn’s excellent article](http://christophermeiklejohn.com/orleans.html).
+
+### Introduction
+
+Different focus:
+
+  * Akka as a toolkit for building distributed systems, offering the full power but also exposing the inherent essential complexity.
+  
+  * Orleans restricts applicability in order to allow seamless use without understanding distributed computing.
+  
+#### Lifecycle
+
+  * Orleans Grains do not have a lifecycle, can neither be started nor stopped. Consequently they also cannot fail and be restarted, therefore Orleans does not offer tools for software failure handling—the failure handling aspects concentrate on recovering from hardware crashes.
+  
+  * Akka Actors implement the full model including defined lifecycle beginning and end, these are explicit operations. Persistent Actors support extending the lifecycle of a logical unit of computation beyond the lifecycle of the running process instance. Restarting an Actor provides powerful means for automatic service recovery.
+  
+#### Automatic Creation
+
+  * Orleans Grains are created automatically whenever needed, which implies that initialization activities cannot have persistent effects—otherwise these would be uncontrollable. OTOH this frees the user from having to consider the need for Grain creation. It also means that asking for a Grain you don’t know which implementation you will get.
```

Comment:
Yes, OnActivateAsync  is not explicit. If you need an explicit control of "Do X", you would write it as grain method.